airflow/jobs/base_job.py

```diff
@@ -245,32 +246,47 @@ def heartbeat(self, only_if_necessary: bool = False):
             # We didn't manage to heartbeat, so make sure that the timestamp isn't updated
             self.latest_heartbeat = previous_heartbeat
 
+    @staticmethod
+    @internal_api_call
+    @provide_session
+    def pre_execute(job: BaseJob, *, session: Session = NEW_SESSION) -> None:
+        """Update the database job entry before running the _execute() function."""
+        job.state = State.RUNNING
+        session.add(job)
+        session.commit()
+        make_transient(job)
```

Comment:
Wondering if this `make_transient` call is necessary in case of execution on the internal api.

Reply:
I think there is a bigger problem here. BaseJob is an SQLAlchemy model and we are effectively trying to serialize it with internal_api call. This is not going to end well.

I think this is one of those cases where we need to do it differently and refactor the code quite a bit deeper.

Reply:
(BTW. This is the first case I see we try to do that, so it's not a general issue so far).

Reply:
Yep, much bigger problem :joy:.

Putting this one in draft as I will not be able to update before going on holidays.